src/shared/components/TextExpander.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import {observer} from "mobx-react";
+import {If, Else, Then} from 'react-if';
+import TextTruncate from 'react-text-truncate';
+import {observable} from "mobx";
+
+export interface ITextExpanderProps
+{
+    text?: string|JSX.Element|null;
+    line?: number;
+    expandIndicator?: string|JSX.Element;
+    collapseIndicator?: string|JSX.Element;
+    truncateText?: string;
+}
+
+@observer
+export default class TextExpander extends React.Component<ITextExpanderProps, {}> {
+    @observable protected isTextTruncated: boolean = true;
```

Comment:
I didn't really realize that we favor `setState` over `observable` in those kind of situations. Recently, I have started using `observable` instead of `setState`.